dandi/cli/command.py

```diff
@@ -0,0 +1,223 @@
+            ))
+        ]
+    )
+    if not sys.stdout.isatty():
+        # TODO: ATM width in the final mode is hardcoded
+        #  https://github.com/pyout/pyout/issues/70
+        # and depending on how it would be resolved, there might be a
+        # need to specify it here as "max" or smth like that.
+        # For now hardcoding to hopefully wide enough 200 if stdout is not
+        # a tty
+        PYOUT_STYLE['width_'] = 200
+
+    out = pyout.Tabular(
+        columns=[
+            'path',
+            'size',
+            #'experiment_description',
+            #'lab',
+            'experimenter',
+            'session_id',
+            'session_start_time',
+            #'identifier',  # note: required arg2 of NWBFile
+            #'institution',
+            'keywords',
+            #'related_publications',
+            #'session_description',  # note: required arg1 of NWBFile
+        ],
+        style=PYOUT_STYLE
+        # , stream=...
+    )
+    with out:
+        for path in files:
+            rec = {'path': path}
+            try:
+                rec['size'] = os.stat(path).st_size
+                rec.update(get_metadata_pyout(path))
+            except FileNotFoundError as exc:
+                #lgr.error("File is not available: %s", exc)
+                pass
+            out(rec)
+        out.wait()
+
+#
+# Validate
+#
+
+@main.command()
+@click.argument('paths', nargs=-1, type=click.Path())
+def validate(paths):
+    """Validate files for NWB (and DANDI) compliance
+
+    Exits with non-0 exit code if any file is not compliant.
+    """
+    files = get_files(paths)
+    import pynwb
+    import crayons
+
+    errors = {}
+    for path in files:
+        try:
+            with pynwb.NWBHDF5IO(path, 'r', load_namespaces=True) as reader:
+                validation = pynwb.validate(reader)
+                if validation:
+                    errors[path] = validation
+        except Exception as exc:
+            errors[path] = ["Failed to validate: %s" % str(exc)]
+
+    # TODO: Most likely we want to summarize errors across files since they
+    # are likely to be similar
+    # TODO: add our own criteria for validation (i.e. having needed metadata)
+    if errors:
+        # # can't be done since fails to compare different types of errors
+        # all_errors = sum(errors.values(), [])
+        # all_error_types = []
+        # errors_unique = sorted(set(all_errors))
+        # from collections import Counter
+        # # Let's make it
+        # print(
+        #     "{} unique errors in {} files".format(
+        #     len(errors_unique), len(errors))
+        # )
+        for path, errors in errors.items():
+            print(crayons.normal(path, bold=True))
+            for error in errors:
+                print("  {}".format(crayons.red(error)))
+        raise SystemExit(1)
+    else:
+        print(
+            crayons.green("No validation errors among {} files".format(len(files)),
```

Comment:
I'm unfamiliar with crayons, does it provide something more than `click.style` http://click.palletsprojects.com/en/7.x/utils/#ansi-colors?

Reply:
oh, I didn't know about that feature of click.  most likely it would be sufficient - I will check. thanks!
I picked up crayons idea from pipenv which also uses click.